scipy/optimize/optimize.py

```diff
@@ -483,7 +482,8 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
 
     iterations = 1
 
-    while (fcalls[0] < maxfun and iterations < maxiter):
+    while ((maxfun is None or fcalls[0] < maxfun)
+        and (maxiter is None or iterations < maxiter)):
```

Comment:
I think @pv was suggesting:

    if maxiter is None and maxfun is None:
        maxiter = N * 200
        maxfun = N * 200
    elif maxiter is not None and maxfun is not None:
        pass
    elif maxiter is not None:
        maxfun = np.inf
    elif maxfun is not None:
        maxiter = np.inf